library/src/main/java/com/chuckerteam/chucker/internal/data/entity/HttpTransaction.kt

```diff
@@ -208,13 +208,27 @@ internal class HttpTransaction(
     }
 
     fun populateUrl(url: HttpUrl): HttpTransaction {
-        this.url = url.toString()
-        host = url.host()
-        path = ("/${url.pathSegments().joinToString("/")}${url.query()?.let { "?$it" } ?: ""}")
-        scheme = url.scheme()
+        val formattedUrl = FormattedUrl.fromHttpUrl(url, encoded = false)
+        this.url = formattedUrl.url
```

Comment:
Maybe we could rename `url` parameter for this function, so there will be no need in using `this.url` for this line?

Reply:
How about `httpUrl`?

Reply:
Works for me 👍